dart/dynamics/Skeleton.h

```diff
@@ -591,8 +606,15 @@ class Skeleton
   //----------------------------------------------------------------------------
   friend class BodyNode;
   friend class Joint;
+  friend class DegreeOfFreedom;
 
 protected:
+  /// Register a joint with the Skeleton. Internal use only.
+  void registerJoint(dart::dynamics::Joint* _newJoint);
```

Comment:
Skeleton and Joint are in the same namespace. Why the full namespace is specified here?